core/test/base/executor.cpp

```diff
@@ -474,6 +474,76 @@ TEST(DpcppExecutor, KnowsItsDeviceId)
 }
 
 
+TEST(Executor, CanVerifyMemory)
+{
+    auto ref = gko::ReferenceExecutor::create();
+    auto omp = gko::OmpExecutor::create();
+    auto hip = gko::HipExecutor::create(0, omp);
+    auto cuda = gko::CudaExecutor::create(0, omp);
+    auto omp2 = gko::OmpExecutor::create();
+    auto hip2 = gko::HipExecutor::create(0, omp);
+    auto cuda2 = gko::CudaExecutor::create(0, omp);
+    auto hip_1 = gko::HipExecutor::create(1, omp);
+    auto cuda_1 = gko::CudaExecutor::create(1, omp);
+    std::shared_ptr<gko::DpcppExecutor> host_dpcpp;
+    std::shared_ptr<gko::DpcppExecutor> cpu_dpcpp;
+    std::shared_ptr<gko::DpcppExecutor> gpu_dpcpp;
+    if (gko::DpcppExecutor::get_num_devices("host"))
+        host_dpcpp = gko::DpcppExecutor::create(0, omp, "host");
+    if (gko::DpcppExecutor::get_num_devices("cpu"))
+        cpu_dpcpp = gko::DpcppExecutor::create(0, omp, "cpu");
+    if (gko::DpcppExecutor::get_num_devices("gpu"))
+        gpu_dpcpp = gko::DpcppExecutor::create(0, omp, "gpu");
```

Comment:
Is it intentional, that the `gpu_dpcpp` executor is an `omp` executor?
Note: Comment only applies to last line.
```suggestion
    if (gko::DpcppExecutor::get_num_devices("host"))
        host_dpcpp = gko::DpcppExecutor::create(0, omp, "host");
    if (gko::DpcppExecutor::get_num_devices("cpu"))
        cpu_dpcpp = gko::DpcppExecutor::create(0, omp, "cpu");
    if (gko::DpcppExecutor::get_num_devices("gpu"))
        gpu_dpcpp = gko::DpcppExecutor::create(0, cuda, "gpu");
```

Reply:
Yes, the same way as with CUDA devices, we want to have access to a real host side executor for doing our Ginkgo copies through when we use an Intel GPU. In fact, such host side executor is not required when creating an `host` or `cpu` DPC++ executor, but we need to have something just in case. Another issue is that you can control what we see through the SYCL environment variables, and it's possible to enforce, up to a point, what device is actually selected from outside Ginkgo's scope...

Reply:
Can you add a comment for this case, to avoid this confusion?

Reply:
I'm not sure what clarification is needed here, the code isn't different in essence from:
```
auto cuda = gko::CudaExecutor::create(0, omp);
```